packages/teams-js/src/public/meeting.ts

```diff
@@ -925,6 +925,124 @@ export namespace meeting {
+    }
+    const { threadId, messageId, replyChainMessageId } = joinMeetingParams.chatInfo;
+
+    joinMeetingParams.joinWebUrl = new URL(joinMeetingParams.joinWebUrl);
+    const serializedJoinMeetingParams = {
+      joinWebUrl: joinMeetingParams.joinWebUrl.href,
+      subject: joinMeetingParams.subject,
+      source: joinMeetingParams.source || EventActionSource.Other,
+      chatInfo: {
+        threadId,
+        messageId: messageId || '0',
+        replyChainMessageId: replyChainMessageId || '0',
+      },
+    };
+    return new Promise<void>((resolve) => {
+      ensureInitialized(runtime);
+      resolve(
+        sendAndHandleStatusAndReason(
+          getApiVersionTag(ApiVersionNumber.V_2, ApiName.Meeting_JoinMeeting),
+          'meeting.joinMeeting',
+          serializedJoinMeetingParams,
+        ),
+      );
+    });
+  }
+
+  /**
+   * This function is used to check the validity of joinMeetingParams which return true if the joinMeetingParams' joinWebUrl and chatInfo are valid else false.
+   * @param joinMeetingParams The parameters for joining the meeting. {@link JoinMeetingParams}
+   * @returns false if joinMeetingParams' joinWebUrl or chatInfo is not valid else true.
+   */
+  function validateJoinMeetingParams(joinMeetingParams: JoinMeetingParams): boolean {
+    if (
+      !joinMeetingParams?.chatInfo?.threadId ||
+      !joinMeetingParams?.joinWebUrl ||
+      joinMeetingParams.chatInfo.threadId === ''
+    ) {
+      return false;
+    }
+    if (typeof joinMeetingParams.joinWebUrl === 'string' && joinMeetingParams.joinWebUrl === '') {
+      return false;
+    }
+
+    if (joinMeetingParams.joinWebUrl instanceof URL && joinMeetingParams.joinWebUrl.href === '') {
+      return false;
+    }
+    return true;
+  }
+
+  /**
+   * Contains information associated with parameters required for joining the Microsoft Teams meetings.
+   * More details can be found at:
+   * [Online Meeting Base - Microsoft Graph v1.0](https://learn.microsoft.com/en-us/graph/api/resources/onlinemeetingbase?view=graph-rest-1.0)
+   */
+  export interface JoinMeetingParams {
+    /** The join URL of the online meeting. */
+    joinWebUrl: URL | string;
+    /** The chat information associated with this online meeting. {@link ChatInfo}*/
+    chatInfo: ChatInfo;
+    /** The subject of the meeting. */
+    subject?: string;
+    /** The source of the join button click. {@link EventActionSource} */
+    source?: EventActionSource;
+  }
+
+  /**
+   * Contains information associated with Microsoft Teams meetings.
+   * More details can be found at:
+   * [Chat Info - Microsoft Graph v1.0](https://learn.microsoft.com/en-us/graph/api/resources/chatinfo?view=graph-rest-1.0)
+   */
+  export interface ChatInfo {
+    /** The unique identifier for a thread in Microsoft Teams. */
+    threadId: string;
+    /** The unique identifier of a message in a Microsoft Teams channel. */
+    messageId?: string;
```

Comment:
In what circumstances would a developer set the `messageId` and `replyChainMessageId`? In a side chat you told me this:

"The threadId (and other information) is needed to open the right side chat during a meeting."

`threadid` I get, it makes sense that you would have to specify what chat thread to open in the sidepane. `messageId` and `replyChainMessageId` I don't understand. Both of them seem like they are specifically used for identifying single messages in that thread, which doesn't seem like something a developer would need to do in order to open the sidepane.

Does that sound correct?

Reply:
@TrevorJoelHarris, developer will have to pass messageId and replyChainMessageId in case of a channel meeting. In case of channel meetings, messageId and replyChannelMessageId are used to find the post in a channel where this meeting resides.